#25 changed library in SendRequestsService class and SendRequestsServ…

klimenkoMA opened the pull request:
Close#25 changed done

klimenkoMA committed:
#25 changed library in SendRequestsService class and SendRequestsServiceTest class,
changed assertions in sendGet() and sendPost()

Commit:
Minor refactoring. Moved some constants to local variables, moved comments, added some missing spaces

spasynkov commented on src/test/java/com/example/telegrambotinsurance/service/SendRequestsServiceTest.java:
```diff
@@ -22,24 +23,24 @@
+	void sendGet() throws JSONException {
 		JSONObject response = sendRequestsService.sendGet(token,"getMe");
 		//Проверка, что ответ не пустой
 		Assertions.assertNotNull(response);
 		//Проверка, что ответ удачный
-		Assertions.assertTrue("true".equals(response.getAsString("ok")),"Wrong response from telegram servers, field: ok = false");
+		Assertions.assertEquals(response.getString("ok"), "true", "Wrong response from telegram servers, field: ok = false");
 	}
 
 	/**
 	 * Проверка ответа от метода sendGet и отправка сообщения "Testing sendPost()"
 	 */
 	@Test
-	void sendPost() {
+	void sendPost() throws JSONException {
 		JSONObject response = sendRequestsService.sendPost(token,"sendMessage",
 				"{\"chat_id\":-1001484722738, \"text\":\"Testing sendPost()\"}");
 		//Проверка, что ответ не пустой
 		Assertions.assertNotNull(response);
 		//Проверка, что ответ удачный
-		Assertions.assertTrue("true".equals(response.getAsString("ok")),"Wrong response from telegram servers, field: ok = false");
+		Assertions.assertEquals(response.getString("ok"), "true", "Wrong response from telegram servers, field: ok = false");
```
опача :)
а зачем тесты трогал?)
когда пишешь основной код - тесты не должны меняться :)
проблема в том, что тестовые методы не должны кидать никаких исключений. если в тесте произошло исключение - значит что-то в нем не работает.
если невозможно достать стрингу - значит тест должен завалиться, а не программа целиком :)
мавен, например, когда билдит проект - он тоже запускает тесты. и билд не должен валиться из-за того, что в тестах **_ПРОВЕРЯЕМОЕ_** исключение не проверили и не обработали :)

klimenkoMA replied:
Принял, исправлю

spasynkov reviewed (changes requested):
тесты не работают, это первое и самое главное

второе - тесты менять не надо было. библиотеку поменять, конечно, но структуру тестов - не надо было :) и особенно нельзя логику тестов подгонять под свой результат (хоть этого и не было, но все же)

klimenkoMA committed:
#25 changes in SendRequestService class
- rebuild sendGet()
- rebuild sendPost()

klimenkoMA committed:
Merge remote-tracking branch 'origin/#25ChangeJSONLibraries' into #25ChangeJSONLibraries

klimenkoMA committed:
#25 changes in SendRequestServiceTest class
- rebuild sendGet() and sendPost(), removed
"trows JSONException", changed chat_id in sendPost()

klimenkoMA committed:
#25 changes in SendRequestServiceTest class
- deleted import JSONException

spasynkov reviewed (approved):
вроде ок
